tools/src/main/java/io/github/ygojson/tools/dataprovider/impl/yugipedia/model/IntegerOrUndefined.java

```diff
@@ -0,0 +1,49 @@
+package io.github.ygojson.tools.dataprovider.impl.yugipedia.model;
+
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.fasterxml.jackson.databind.ser.std.ToStringSerializer;
+
+/**
+ * Identifies an integer that can be undefined by matching '?'.
+ */
+@JsonSerialize(using = ToStringSerializer.class)
+public class IntegerOrUndefined {
```

Comment:
Add tests to define the behavior that we want with different kind of values and adapt as expected.